people-and-planet-ai/timeseries-classification/main.py

```diff
@@ -0,0 +1,153 @@
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from datetime import datetime
+import os
+
+import flask
+import numpy as np
+
+app = flask.Flask(__name__)
+
+# Default values for dataset creation.
+DEFAULT_TRAIN_EVAL_SPLIT = [80, 20]
+
+# Default values for training in Vertex AI.
+DEFAULT_TRAIN_EPOCHS = 100
+DEFAULT_BATCH_SIZE = 128
+DEFAULT_MACHINE_TYPE = "n1-standard-4"
+DEFAULT_GPU_TYPE = "NVIDIA_TESLA_T4"
+DEFAULT_GPU_COUNT = 2
+
+# Google Cloud resources.
+PROJECT = os.environ["PROJECT"]
+REGION = os.environ["REGION"]
+STORAGE_PATH = os.environ["STORAGE_PATH"]
+CONTAINER_IMAGE = os.environ["CONTAINER_IMAGE"]
+
+RAW_DATA_DIR = f"{STORAGE_PATH}/data"
+RAW_LABELS_DIR = f"{STORAGE_PATH}/labels"
+TRAIN_DATA_DIR = f"{STORAGE_PATH}/datasets/train"
+EVAL_DATA_DIR = f"{STORAGE_PATH}/datasets/eval"
+TRAINING_DIR = f"{STORAGE_PATH}/training"
+TEMP_DIR = f"{STORAGE_PATH}/temp"
+
+
+@app.route("/ping", methods=["POST"])
+def run_root() -> dict:
+    args = flask.request.get_json() or {}
+    return {
+        "response": "Your request was successful! 🎉",
+        "args": args,
+    }
+
+
+@app.route("/create-datasets", methods=["POST"])
+def run_create_datasets() -> dict:
+    import create_datasets
+
+    try:
+        args = flask.request.get_json() or {}
+        runner_params = {
+            "runner": "DataflowRunner",
+            "job_name": f"global-fishing-watch-create-datasets-{datetime.now().strftime('%Y%m%d-%H%M%S')}",
+            "save_main_session": args.get("save_main_session", True),
+            "sdk_container_image": args.get("container_image", CONTAINER_IMAGE),
+            "project": args.get("project", PROJECT),
+            "region": args.get("region", REGION),
+            "temp_location": args.get("temp_location", TEMP_DIR),
+            "experiments": ["use_runner_v2"],
+        }
+        params = {
+            "raw_data_dir": args.get("raw_data_dir", RAW_DATA_DIR),
+            "raw_labels_dir": args.get("raw_labels_dir", RAW_LABELS_DIR),
+            "train_data_dir": args.get("train_data_dir", TRAIN_DATA_DIR),
+            "eval_data_dir": args.get("eval_data_dir", EVAL_DATA_DIR),
+            "train_eval_split": args.get("train_eval_split", DEFAULT_TRAIN_EVAL_SPLIT),
+            **runner_params,
+        }
+        job_id = create_datasets.run(**params)
+
+        return {
+            "method": "create-datasets",
+            "job_id": job_id,
+            "job_url": f"https://console.cloud.google.com/dataflow/jobs/{REGION}/{job_id}?project={PROJECT}",
+            "params": params,
+        }
+    except Exception as e:
```

Comment:
Can this exception be made more specific? Is it possible for the try/except block to be smaller?

https://google.github.io/styleguide/pyguide.html#244-decision

Reply:
Great point, I'm also not a fan of open exceptions like this, but in this case I think it falls under one of the exceptions:

```
Never use catch-all except: statements, or catch Exception or StandardError, unless you are

* re-raising the exception, or
* creating an isolation point in the program where exceptions are not propagated but are
  recorded and suppressed instead, such as protecting a thread from crashing by guarding
  its outermost block.
```

In this case, all the exception catching in `main.py` is surrounding the entire underlying call to avoid having the server instance crash, and so we actually get a more sensible error message in the client side other than a 500 error. Ideally, each underlying call should handle its more specific exceptions.